src/test/java/org/mule/test/http/internal/listener/HttpResponseFactoryTestCase.java

```diff
@@ -83,4 +88,52 @@ public void testCursorIteratorCausesInvalidTypeForHttpResponseError() throws Run
     httpResponseBuilder.create(HttpResponse.builder(), new NoInterception(), listenerResponseBuilder, true);
   }
 
+  @Test
+  @Description("Forces connection close header honoring the constructor supplier.")
+  public void forceConnectionCloseHeaderWhenNotPresent() {
+    checkForceConnectionCloseHeader(true, null);
+  }
+
+  @Test
+  @Description("Forces connection close header honoring the constructor supplier.")
+  public void forceConnectionCloseHeaderWhenPresent() {
+    checkForceConnectionCloseHeader(true, KEEP_ALIVE);
+  }
+
+  @Test
+  @Description("Doesn't force connection close header honoring the constructor supplier.")
+  public void doesNotForceConnectionCloseHeaderWhenNotPresent() {
+    checkForceConnectionCloseHeader(false, null);
+  }
+
+  @Test
+  @Description("Doesn't force connection close header honoring the constructor supplier.")
+  public void doesNotForceConnectionCloseHeaderWhenPresent() {
+    checkForceConnectionCloseHeader(false, KEEP_ALIVE);
+  }
+
+  private void checkForceConnectionCloseHeader(final boolean shouldForceConnectionClose, String defaultValue) {
+    HttpListenerResponseBuilder listenerResponseBuilder = mock(HttpListenerResponseBuilder.class);
+    TypedValue<Object> payload = new TypedValue<>(EXAMPLE_STRING, STRING);
+    when(listenerResponseBuilder.getBody()).thenReturn(payload);
+    when(listenerResponseBuilder.getStatusCode()).thenReturn(OK.getStatusCode());
+
+    MultiMap<String, String> headers = new MultiMap<>();
+    if (defaultValue != null) {
+      headers.put(CONNECTION, defaultValue);
```

Comment:
you can use Collections.singletonMap() for this

Reply:
But I only want to set the connections header when defaultValue != null; Otherwise the map should be empty

Reply:
you have `Collections.emptyMap()` for that

Reply:
Trying to apply the change, I found it's a MultiMap